src/text-editor-presenter.coffee

```diff
@@ -94,40 +82,68 @@ class TextEditorPresenter
     @updateCustomGutterDecorationState() if @shouldUpdateCustomGutterDecorationState
     @updating = false
 
+    @resetTrackedUpdates()
+
     @state
 
+  resetTrackedUpdates: ->
+    @shouldUpdateFocusedState = false
+    @shouldUpdateHeightState = false
+    @shouldUpdateVerticalScrollState = false
+    @shouldUpdateHorizontalScrollState = false
+    @shouldUpdateScrollbarsState = false
+    @shouldUpdateHiddenInputState = false
+    @shouldUpdateContentState = false
+    @shouldUpdateDecorations = false
+    @shouldUpdateLinesState = false
+    @shouldUpdateCursorsState = false
+    @shouldUpdateOverlaysState = false
+    @shouldUpdateLineNumberGutterState = false
+    @shouldUpdateLineNumbersState = false
+    @shouldUpdateGutterOrderState = false
+    @shouldUpdateCustomGutterDecorationState = false
+
   observeModel: ->
     @disposables.add @model.onDidChange =>
       @updateContentDimensions()
       @updateEndRow()
-      @updateHeightState()
-      @updateVerticalScrollState()
-      @updateHorizontalScrollState()
-      @updateScrollbarsState()
-      @updateContentState()
-      @updateDecorations()
-      @updateLinesState()
-      @updateLineNumberGutterState()
-      @updateLineNumbersState()
-      @updateGutterOrderState()
-      @updateCustomGutterDecorationState()
+      @shouldUpdateHeightState = true
```

Comment:
> What are some of the other batched cases you're thinking of?

@nathansobo: `DisplayBuffer` and `TextEditorPresenter`, for example, defer some operations whenever character widths are being changed (https://github.com/atom/atom/blob/master/src/display-buffer.coffee#L307-L312).

> I tend to agree with @benogle that it's hard to beat the straightforwardness of a list of subroutines that get called conditionally. I worry that it might be another case of an abstraction that dries things up but also adds some opacity.

> If we're going to get more dynamic, there's lots more we could do. For example, could we somehow express the dependencies between various pieces of state to derive the list of what needs to update automatically, like a spreadsheet. That would make things less error prone, but also less explicit. I'm not sure.

Agreed to keep things simple right now: booleans clearly express the intent of deferring work and I think we should favor explicitness over duplication removal if it causes code to be less clear.